### PR TITLE
ff: fix rewriter

### PR DESCRIPTION
**Pull request by alex-ozdemir**
I'm not sure combining like terms can be expressed in a single pass (without fully flattening)

Thanks for setting up the fuzzer, Mathias!

fixes https://github.com/cvc5/cvc5-projects/issues/704

### CODE REVIEW

File: test/regress/cli/CMakeLists.txt [hunk 1]
```diff
@@ -777,6 +777,7 @@ set(regress_0_tests
   regress0/ff/bool_nary_or_unsound.smt2
   regress0/ff/combination.smt2
   regress0/ff/ctx.smt2
+  regress0/ff/issue704.smt2
```
Comment (mpreiner) [inline test/regress/cli/CMakeLists.txt:780]:
```suggestion
  regress0/ff/proj-issue704.smt2
```

Comment (mpreiner) [reply test/regress/cli/CMakeLists.txt:780]: Make sure to rename the regression test accordingly :)

File: test/regress/cli/regress0/ff/issue704.smt2 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(set-option :check-models true)
+(declare-const x (_ FiniteField 13))
```
Comment (mpreiner) [inline test/regress/cli/regress0/ff/issue704.smt2:2]: Can you add `set-logic` and `set-info :status` commands?

**Comment by alex-ozdemir** [comment]
Thanks for the comments!

File: test/regress/cli/regress0/ff/proj-issue704.smt2 [hunk 3]
```diff
@@ -0,0 +1,8 @@
+; SCRUBBER: grep -w 'sat'
```
Comment (mpreiner) [inline test/regress/cli/regress0/ff/proj-issue704.smt2:1]: Is this line actually required if the status is set? The regression script should be able to check the output.

Comment (alex-ozdemir) [reply test/regress/cli/regress0/ff/proj-issue704.smt2:1]: This line trims the output of `get-value`. I expect that the exact form of the output is unstable, so I didn't want to check for it.

**Review by mpreiner — APPROVED** [review]

**Pull request merged**